pkg/arvo/lib/lm.hoon

```diff
@@ -0,0 +1,702 @@
+::
+::::  Matrix type in single-precision floating-point @rs
+  ::
+/+  *lv
+|%
+++  lms
+  ^|
+  ~%  %lms  ..is  ~
+  |_  r/$?($n $u $d $z)   :: round nearest, round up, round down, round to zero
+  ::
+  ::  Manipulators
+  ::
+  ::    Cantor pairing function
+  ::      This is used so that the dimensions can be stored as a unique number.
+  ++  cantor
+    ~/  %cantor
+    |=  [m=@ud n=@ud]  ^-  @ud
+    (add (div (mul (add m n) (add m (add n 1))) 2) n)
+  ++  decantor
+    ~/  %decantor
+    |=  [z=@ud]  ^-  (list @ud)
+    =/  w  (div (sub p:(sqt (add (mul 8 z) 1)) 1) 2)
+    =/  t  (div (add (mul w w) w) 2)
+    =/  y  (sub z t)
+    =/  x  (sub w y)
+    ~[x y]
```

Comment:
Honestly I'd just use `(rep 4 [m n]~)` here, cantor of two `@ud` into a `u32` gives you <2x range of just having a pair of u16, and outright less if your matrices are anywhere near square.

Reply:
Adopted throughout.